livre/core/data/Histogram.cpp

```diff
@@ -0,0 +1,127 @@
+/* Copyright (c) 2011-2015, EPFL/Blue Brain Project
+ *                     Ahmet Bilgili <ahmet.bilgili@epfl.ch>
+ *
+ * This file is part of Livre <https://github.com/BlueBrain/Livre>
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License version 3.0 as published
+ * by the Free Software Foundation.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this library; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+#include <livre/core/data/Histogram.h>
+
+namespace livre
+{
+
+Histogram::Histogram()
+    : _minIndex( -1u )
+    , _maxIndex( -1u )
+{
+    std::memset( getBins(), 0, sizeof( uint64_t ) * binCount );
+}
+
+Histogram::Histogram( const Histogram& histogram )
+    : co::Distributable< ::lexis::render::ViewHistogram >( histogram )
+{
+    if( this == &histogram )
+        return;
+
+    *this = histogram;
+}
+
+Histogram& Histogram::operator=( const Histogram& histogram )
+{
+    if( this == &histogram )
+        return *this;
+
+    const uint64_t* srcBins = histogram.getBins();
+    uint64_t* bins = getBins();
+    std::memcpy( bins, srcBins, sizeof( uint64_t) * binCount );
+    setId( histogram.getId( ));
+    setArea( histogram.getArea( ));
+    _minIndex = -1u;
+    _maxIndex = -1u;
+    return *this;
+}
+
+bool Histogram::isComplete() const
+{
+    const float eps = 0.0001f;
+    return std::abs( 1.0f - getArea( )) <= eps;
+}
+
+Histogram& Histogram::operator+=( const Histogram& histogram )
+{
+    if( this == &histogram )
+        return *this;
```

Comment:
This is not correct. Not that this code is often written, but compare the result you would expect from this:

  int a = 2
  a += a

to your code.